zjit/src/options.rs

```diff
@@ -235,6 +241,10 @@ fn parse_option(str_ptr: *const std::os::raw::c_char) -> Option<()> {
             options.print_stats = false;
         }
 
+        ("trace-exits", "") => {
+            options.trace_side_exits = true;
```

Comment:
Does it make sense to also set `options.stats = true;` here so `--zjit-stats` doesn't need to specified separately?
I feel if it's a hard dependency we can just enable both of them together.

Reply:
As I raised at https://github.com/ruby/ruby/pull/14640#discussion_r2380501403 (which wasn't really addressed for the whole patch), I still wonder if we really need the dependency between them in the first place. 

If it just relies on `rb_zjit_record_exit_stack` calls and it works without stats, we should be able to just get rid of `get_option!(stats)` guards instead of letting `--zjit-trace-exits` depend on `--zjit-stats`.